openshift/e2e-tests-openshift.sh

```diff
@@ -254,6 +283,50 @@ function run_e2e_tests(){
   return $failed
 }
 
+function run_rolling_upgrade_tests() {
+    header "Running rolling upgrade tests"
+    
+    local TIMEOUT_TESTS="20m"
+    failed=0
+
+    report_go_test -tags=preupgrade -timeout=${TIMEOUT_TESTS} ./test/upgrade \
+    --dockerrepo "${INTERNAL_REGISTRY}/${SERVING_NAMESPACE}" \
+    --kubeconfig $KUBECONFIG || failed=1
+
+    echo "Starting prober test"
+
+    # Make prober send requests more often compared with upstream where it is 1/second
+    sed -e 's/\(.*requestInterval =\).*/\1 200 * time.Millisecond/' -i vendor/knative.dev/pkg/test/spoof/spoof.go
+
+    rm -f /tmp/prober-signal
+    report_go_test -tags=probe -timeout=${TIMEOUT_TESTS} ./test/upgrade \
+    --dockerrepo "${INTERNAL_REGISTRY}/${SERVING_NAMESPACE}" \
+    --kubeconfig $KUBECONFIG &
+
+    PROBER_PID=$!
+    echo "Prober PID is ${PROBER_PID}"
+
+    local serving_version=$(oc get knativeserving knative-serving -o=jsonpath="{.status.version}")
+
+    local latest_csv=$(oc get configmaps serverless-operator -n $OLM_NAMESPACE -o jsonpath='{.data.packages}' | grep currentCSV | awk '{print $2}')
+    approve_csv $latest_csv || failed=1
+    
+    # Wait for the new KnativeServing to be ready
+    timeout 900 '[[ $(oc get knativeserving knative-serving -o=jsonpath="{.status.version}") == $serving_version || $(oc get knativeserving knative-serving -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") != True ]]' || return 1
```

Comment:
I don't quite understand the condition here. Why would we wait for the version to be the same as before the upgrade (we read the version before approving the CSV).

Reply:
Well. That's because the "timeout" function uses `while` instead of `until`. So I have to revert the condition. In fact I'm waiting for: .status.version != serving_version AND Ready=True

Reply:
oh yeah. I remember. I hate it 😂 .

Reply:
Nothing for you to change though.